README.md

```diff
@@ -0,0 +1,569 @@
+        unset($oBucket);
+
+        return parent::PASS_ON;
+    }
+}
+```
+
+Great. Now let's register our filter under a specific name:
+
+```php
+$filterName = 'toupper';
+Hoa\Stream\Filter::register($filterName, ToUpper::class);
+```
+
+Then, we must apply the filter on a specific stream, so let's open a
+stream, and append the filter:
+
+```php
+$file = new Hoa\File\Read(__FILE__);
+Hoa\Stream\Filter::append($file, $filterName, Hoa\Stream\Filter::READ);
+```
+
+This filter has been applied for reading operations only. So we will
+see its effect when reading on our stream, let's do it:
+
+``` php
+echo $file->readAll();
+```
+
+You will see everything in ASCII uppercase.
+
+A filter is a low-level stream API. It integrates with all kind of
+streams. And this is a very powerful tool. We mentionned some usages
+like decrypt, transform to, unzip… Actually, PHP comes with certain
+standard filters, like: `string.toupper`, `string.tolower`, `dechunk`,
+`zlib.*`, `bzip2.*`, `convert.iconv.*` etc. The
+`Hoa\Stream\Filter\Filter::getRegistered` method will provide the list
+of all registered filters.
+
+The `Hoa\Stream\Filter\LateComputed` class is a special filter. It
+calls its public `compute` method when the stream reaches its end. So
+by extending this filter, you can override the `compute` method and
+works on the `_buffer` attribute. This buffer contains the whole
+content of the stream. This is really a buffer. Why would it be
+useful? For instance if you are reading a PHP file, you can transform
+the source code on-the-fly by using a parser —for instance— and
+rewrite parts of the file. This technique is particularily useful to
+instrument codes (adding some probes).
+
+This is also possible to auto-apply a filter with… a wrapper! For
+example the `instrument://` wrapper can prepend a filter to the stream
+being opened with the `stream_open` method (from the
+`Hoa\Stream\Wrapper\IWrapper\Stream` interface).
+
+Possibilities are numerous.
+
+### Other operations
+
+There are more to cover. `Hoa\Stream` supports composite streams (with
+the `Hoa\Stream\Composite` abstract class), i.e. streams embedding
+other streams, like
+[the `Hoa\Xml` library](https://central.hoa-project.net/Resource/Library/Xml).
+An XML stream reads and writes from another inner stream (a file, a
+socket, or anything else).
+[The `Hoa\Stringbuffer` library](https://central.hoa-project.net/Resource/Library/Stringbuffer)
+allows a string to be manipulated with a stream API, so the stream
+content is written on the disk. Stream capabiilities are not the same
+than `Hoa\File` as you might guess.
+
+## Documentation
+
+The
+[hack book of `Hoa\Stream`](https://central.hoa-project.net/Documentation/Library/Stream) contains
+detailed information about how to use this library and how it works.
+
+To generate the documentation locally, execute the following commands:
+
+```sh
+$ composer require --dev hoa/devtools
+$ vendor/bin/hoa devtools:documentation --open
```

Comment:
I think you need to specify to use this section only if it's available to avoid this kink of error `Hoa\Devtools\Bin\Documentation::main(): (1) Directory /tmp/Stream does not contain documentation.
in /tmp/Stream/vendor/hoa/devtools/Bin/Documentation.php at line 166.`

So we need here to add infos to tell it's not yet available.